src/RpcServer/RpcServer.Blockchain.cs

```diff
@@ -197,6 +197,58 @@ protected virtual JToken GetStorage(JArray _params)
             return Convert.ToBase64String(item.Value.Span);
         }
 
+        [RpcMethod]
+        protected virtual JToken FindStorage(JArray _params)
+        {
+            using var snapshot = system.GetSnapshot();
+            if (!int.TryParse(_params[0].AsString(), out int id))
+            {
+                UInt160 hash = UInt160.Parse(_params[0].AsString());
+                ContractState contract = NativeContract.ContractManagement.GetContract(snapshot, hash);
+                if (contract is null) throw new RpcException(-100, "Unknown contract");
```

Comment:
If the id is int and invalid contract, we don't get this error

Reply:
@shargon this is literally a copy/paste from the existing `GetStorage()` implementation
https://github.com/neo-project/neo-modules/blob/99ffc846e4df93491d1d812cb4c6ef6ee875e056/src/RpcServer/RpcServer.Blockchain.cs#L180-L189

Ultimately it will not find any storage. What do you want it to be?

Reply:
If _params[0] is "blah" the application crashes. Cause of this line 185
```csharp
UInt160 hash = UInt160.Parse(_params[0].AsString());
```
crashes the function do to no checking Use UInt160.TryParse

Reply:
> If _params[0] is "blah" the application crashes. Cause of this line 185
> 
> ```cs
> UInt160 hash = UInt160.Parse(_params[0].AsString());
> ```
> 
> crashes the function do to no checking Use UInt160.TryParse

This pattern is in nearly all current RpcMethod's, just look at for example `GetBlock`, `GetBlockHash`, `GetBlockHeader` or `GetRawTransaction`. I'm not saying I disagree this could be handled better, but I'm simply sticking to the current style already in the rest of the RpcServer code unless some maintainer tells me otherwise. I prefer to have this method included in the 3.6.0 release and see more graceful handling be done in a a separate PR for all methods. Could probably be done at the same time as updating it to follow https://github.com/neo-project/proposals/pull/156

Reply:
I understand completely. Not saying your wrong; you bring up a good point, but the way I look at it is that RPC Methods can be in any plugin. For the other plugins they could follow this convention or not. Just saying....